Compilation error with gcc 4.9.4: "error: unknown option after ‘#pragma GCC diagnostic’ kind"

<!--
  Fill in the placeholders below. Delete any headings and placeholders that you do not fill in.
-->
**OS:** Gentoo Linux x86_64
**Version:** gcc 4.9.4
**Commit/Build:** v0.1.0

<!-- Explanation of the issue -->
Compiling, I get the following errors:
```
In file included from /var/tmp/portage/games-simulation/openrct2-0.1.0/work/OpenRCT2-0.1.0/src/openrct2/drawing/X8DrawingEngine.cpp:27:0:
/var/tmp/portage/games-simulation/openrct2-0.1.0/work/OpenRCT2-0.1.0/src/openrct2/drawing/X8DrawingEngine.h:63:32: error: unknown option after ‘#pragma GCC diagnostic’ kind [-Werror=pragmas]
 #pragma GCC diagnostic ignored "-Wsuggest-final-types"
                                ^
/var/tmp/portage/games-simulation/openrct2-0.1.0/work/OpenRCT2-0.1.0/src/openrct2/drawing/X8DrawingEngine.cpp:137:32: error: unknown option after ‘#pragma GCC diagnostic’ kind [-Werror=pragmas]
 #pragma GCC diagnostic ignored "-Wsuggest-final-methods"
                                ^
cc1plus: all warnings being treated as errors
```

These options are not supported by gcc 4.9.4 yet...

--- a/src/openrct2/CMakeLists.txt
+++ b/src/openrct2/CMakeLists.txt
@@ -265,3 +265,13 @@ if (NOT DISABLE_RCT2)
     # builds without need for -fno-omit-frame-pointer
     set_source_files_properties(${CMAKE_CURRENT_LIST_DIR}/rct2/addresses.c PROPERTIES COMPILE_FLAGS -O0)
 endif ()
+
+if (CXX_WARN_SUGGEST_FINAL_TYPES)
+    # Disable -Wsuggest-final-types via pragmas where due.
+   add_definitions(-D__WARN_SUGGEST_FINAL_TYPES__)
+endif ()
+
+if (CXX_WARN_SUGGEST_FINAL_METHODS)
+    # Disable -Wsuggest-final-methods via pragmas where due.
+   add_definitions(-D__WARN_SUGGEST_FINAL_METHODS__)
+endif ()

--- a/src/openrct2/core/Guard.hpp
+++ b/src/openrct2/core/Guard.hpp
@@ -23,7 +23,7 @@ extern "C" {
 #endif // __cplusplus
 
 void openrct2_assert_va(bool expression, const char * message, va_list va);
-inline void openrct2_assert(bool expression, const char * message, ...)
+static inline void openrct2_assert(bool expression, const char * message, ...)
 {
     if (!expression) return;
     va_list va;

--- a/src/openrct2/drawing/X8DrawingEngine.cpp
+++ b/src/openrct2/drawing/X8DrawingEngine.cpp
@@ -133,8 +133,10 @@ void X8RainDrawer::Restore()
     }
 }
 
+#ifdef __WARN_SUGGEST_FINAL_METHODS__
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wsuggest-final-methods"
+#endif
 
 X8DrawingEngine::X8DrawingEngine(Ui::IUiContext * uiContext)
 {
@@ -468,7 +470,9 @@ void X8DrawingEngine::DrawDirtyBlocks(uint32 x, uint32 y, uint32 columns, uint32
     window_draw_all(&_bitsDPI, left, top, right, bottom);
 }
 
+#ifdef __WARN_SUGGEST_FINAL_METHODS__
 #pragma GCC diagnostic pop
+#endif
 
 X8DrawingContext::X8DrawingContext(X8DrawingEngine * engine)
 {

--- a/src/openrct2/drawing/X8DrawingEngine.h
+++ b/src/openrct2/drawing/X8DrawingEngine.h
@@ -66,8 +66,10 @@ namespace OpenRCT2
             void Restore();
         };
 
+#ifdef __WARN_SUGGEST_FINAL_TYPES__
     #pragma GCC diagnostic push
     #pragma GCC diagnostic ignored "-Wsuggest-final-types"
+#endif
         class X8DrawingEngine : public IDrawingEngine
         {
         protected:
@@ -119,7 +121,9 @@ namespace OpenRCT2
             void DrawAllDirtyBlocks();
             void DrawDirtyBlocks(uint32 x, uint32 y, uint32 columns, uint32 rows);
         };
+#ifdef __WARN_SUGGEST_FINAL_TYPES__
     #pragma GCC diagnostic pop
+#endif
 
         class X8DrawingContext final : public IDrawingContext
         {

